### PR TITLE
CentOS7 compatibility

### PR DESCRIPTION
**Pull request by sun77**
- CentOS7 compatibiliy 
  - Use poise-service to manage both systemd and upstart
  - Update metadata
  - Update Cheffile
- Delete useless templates and config file
- Update kitchen.yml
- Correct global_options error with kitchen
  - Add new attribute debug_options
  - Correct template file
  - Correct kitchen.yml

@aboten

### CODE REVIEW

**Comment by kamaradclimber** [comment]
lgtm

**Comment by kamaradclimber** [comment]
please also make a PR to the upstream repo

**Comment by sun77** [comment]
yes of course @kamaradclimber . Just want some internal review before ;)

**Comment by FlorentFlament** [comment]
+1

File: recipes/default.rb [hunk 1]
```diff
@@ -104,14 +104,18 @@
...
     :defaults_timeouts => haproxy_defaults_timeouts
   )
 end
 
-service "haproxy" do
-  supports :restart => true, :status => true, :reload => true
-  action [:enable, :start]
+haproxy_command = ::File.join(node['haproxy']['global_prefix'], 'sbin', 'haproxy')
+poise_service "haproxy" do
+  command "#{haproxy_command} -f #{node['haproxy']['conf_dir']}/haproxy.cfg"
```
Comment (jmauro) [inline recipes/default.rb]: Why doing `::File.join` above and not for this path?

**Comment by jmauro** [comment]
Why removing the init script in case the user doesn't want to use upstart?

File: .kitchen.yml [hunk 2]
```diff
@@ -30,20 +34,30 @@ platforms:
       haproxy:
         test:
           netcat_package: netcat
+- name: debian-6.0.8
```
Comment (jorisd) [inline .kitchen.yml]: You should test debian 7 and debian 8.

Debian 6 is near obsolete now.

**Comment by jmauro** [comment]
:+1:

**Pull request merged**